### ISSUE TITLE
`link_to` does not delete a song in newer versions of Rails, unless jQuery is used.

### ISSUE THREAD

**Issue by Sdcrouse** [learn-co-curriculum/crud-with-validations-lab#26]
Hi. One problem that I had with this lab is that I tried to use `link_to` to delete a song from the index page and the show page. Here's the code that I used:
```
<%= link_to 'Delete song', song, method: :delete, 
data: { confirm: 'Are you sure you want to delete this song?' } %>
```
This generated `<a>` elements such as:
```
<a data-confirm="Are you sure you want to delete this song?" rel="nofollow" 
data-method="delete" href="/songs/3">Delete song</a>
```
I had my routes set up correctly, and I had a corresponding `destroy` action in my SongsController. Despite all of that, every time I clicked on "Delete song", it redirected to the song's show page instead of going to the `destroy` action, and it did not show that "Are you sure you want to delete this song?" message.

After I screen-shared with three Technical Coaches, we finally figured out that newer versions of Rails (5 and above, I think) use jQuery to make all of this work. Without jQuery, the `<a>` element generated by `link_to` goes to the `show` action by default, instead of `destroy`.

**In order to solve this problem, we had to do three things:**

1. Add `gem 'jquery-rails'` to the Gemfile and run `bundle`.
2. Create an `/app/assets/javascripts/application.js` file and add this code to it:
```
//= require jquery
//= require jquery_ujs
//= require_tree .
```
(Note: It's possible that only the second line of code is what's needed, from what I've read online.)

3. Add this line of code to the `/app/views/layouts/application.html.erb` file:
```
  <%= javascript_include_tag 'application.js' %>
```
(In my case, we put it inside of the `<head>` tag, below the `stylesheet_link_tag`.)

After we did all that, the lab worked as it should. Here's the Stack Overflow article that helped us: https://stackoverflow.com/questions/12997167/ruby-on-rails-link-to-delete-method-not-working

Long story short, the README and/or lab itself needs to be updated. Thanks for looking into this!

---Sdcrouse

**Comment by Sdcrouse** [comment]
Update: The `solution` branch of this lab has commits that seem to address this very issue: https://github.com/learn-co-curriculum/crud-with-validations-lab/commit/d190e2703914b882aeea651051827ae8a8d1b026 and https://github.com/learn-co-curriculum/crud-with-validations-lab/commit/b30492f21539de7ad08001068e2b858a2a1ff0d9

But I don't know whether these changes got added to the `master` branch.

**Comment by rrcobb** [comment]
I *think* this is resolved by #16, but let me know if there are any outstanding issues.

**Comment by Sdcrouse** [comment]
@rrcobb Will do! I looked at those merged commits, and I think you're right. I'll be sure to test it out.
